src/Endpoint.ts

```diff
@@ -0,0 +1,186 @@
+
+                let substitutionCost = 0;
+                if (endpoint[i - 1] !== input[j - 1]) {
+                    substitutionCost = 1;
+                }
+
+                // deal with swapped chars
+                if (i > 1 && j > 1 && endpoint[i - 1] === input[j - 2] && endpoint[i - 2] === input[j - 1]) {
+                    d[i][j] = Math.min(
+                            d[i - 1][j] + 1, // delete
+                            d[i][j - 1] + 1, // insertion
+                            d[i - 1][j - 1] + substitutionCost, // substitution
+                            d[i - 2][j - 2] + 1 // transposition
+                        );
+                } else {
+                    d[i][j] = Math.min(
+                            d[i - 1][j] + 1, // deletion
+                            d[i][j - 1] + 1, // insertion
+                            d[i - 1][j - 1] + substitutionCost // substitution
+                        );
+                }
+
+            }
+        }
+
+        const result = d[endpoint.length][input.length];
+        return result;
+    }
+
+    /**
+     *
+     * @param endpoint
+     * @param compareString
+     * @returns minimum distance between endpoint/endpoint synonym and compare string
+     */
+    private calculateEndpointAndSynonymsDist(endpoint: string, argsString: string): number {
+        const endpointComponents = endpoint.split("-");
+        const version = endpointComponents[endpointComponents.length - 1];
+        let compareString = argsString;
+        if (!argsString.endsWith(version)) {
+            compareString = argsString + "-" + version;
+        }
+
+        let dist = this.calculateDistance(endpoint, compareString);
+
+        if (endpoint.startsWith("lol")) {
+            const aliasDist = this.calculateDistance(endpoint.replace("lol", ""), compareString);
+            dist = Math.min(dist, aliasDist);
+        }
+
+        // check for manually defined aliases
+        if (!this.aliases || !this.aliases[endpoint]) {
+            return dist;
+        }
+
+        for (const alias of this.aliases[endpoint]) {
+            const aliasDist = this.calculateDistance(alias + "-" + version, compareString);
+            dist = Math.min(dist, aliasDist);
+        }
+        return dist;
+    }
+
+    private async updateEndpoints() {
+        try {
+            const response = await fetch(`http://www.mingweisamuel.com/riotapi-schema/openapi-3.0.0.json`, {
+                method: "GET",
+                headers: {
+                    "Accept": "application/json",
+                    "Content-Type": "application/json"
+                }
+            });
+
+            if (response.status !== 200) {
+                console.error("HTTP Error trying to get schema: " + response.status);
+                return;
+            }
+
+            const schema = await response.json();
+            const paths = schema.paths;
+
+            const endpointSet = new Set<EndpointName>();
```

Comment:
Could you explain Set to me?

Reply:
I don't want any duplicate endpoints. (i.e. if an endpoint has two routes (https://developer.riotgames.com/api-methods/#champion-v3) I get `champion` twice in the loop below.
I have to copy to an array because fileBackedObject does not properly serialize sets.